File: frontend/heartmage-candles/src/typesV2/constructor/CandleDetail.ts
import { Candle, Decor, LayerColor, NumberOfLayer, Smell, Wick } from '../shared/BaseProduct';

export interface CandleDetail {
  candle: Candle;
  decors?: Decor[];
  layerColors: LayerColor[];
  numberOfLayers: NumberOfLayer[];
  smells?: Smell[];
  wicks: Wick[];
}

File: backend/HeartmadeCandles/HeartmadeCandles.Order.Core/Models/CandleDetail.cs
namespace HeartmadeCandles.Order.Core.Models;

public class CandleDetail
{
    public CandleDetail(
        Candle candle,
        Decor? decor,
        LayerColor[] layerColors,
        NumberOfLayer numberOfLayer,
        Smell? smell,
        Wick wick)
    {
        Candle = candle;
        Decor = decor;
        LayerColors = layerColors;
        NumberOfLayer = numberOfLayer;
        Smell = smell;
        Wick = wick;
    }

    public Candle Candle { get; private set; }
    public Decor? Decor { get; private set; }
    public LayerColor[] LayerColors { get; private set; }
    public NumberOfLayer NumberOfLayer { get; private set; }
    public Smell? Smell { get; private set; }
    public Wick Wick { get; private set; }
}